query/graphql/src/main/java/org/codice/ddf/admin/graphql/transform/GraphQLTransformCommons.java

```diff
@@ -38,6 +46,36 @@ public GraphQLTransformCommons() {
         return transformOutput.fieldsToGraphQLFieldDefinition(Arrays.asList(provider));
     }
 
+    public GraphQLQueryProvider getErrorCodesQueryProvider(List<FieldProvider> fieldProviders) {
+        Set<String> errorCodes = new TreeSet<>();
+
+        for (FieldProvider fieldProvider : fieldProviders) {
+            List<FunctionField> mutations = fieldProvider.getMutationFunctions();
+            List<Field> queryFields = fieldProvider.getDiscoveryFields();
+
+            for (FunctionField mutation : mutations) {
+                errorCodes.addAll(mutation.getErrorCodes());
+            }
+
+            for (Field field : queryFields) {
+                errorCodes.addAll(field.getErrorCodes());
+            }
+        }
+
+        GraphQLEnumType.Builder enumTypeBuilder = GraphQLEnumType.newEnum()
+                .name("ErrorCode")
+                .description("All possible error codes.");
+        errorCodes.forEach(enumTypeBuilder::value);
+        GraphQLEnumType errorCodeEnumType = enumTypeBuilder.build();
+
+        return () -> Collections.singletonList(GraphQLFieldDefinition.newFieldDefinition()
```

Comment:
I don't think it's worth making this with our internal api. It would have to take a reference list to all the field providers. Any time another field provider changed, it would fire an event, triggering the schema to update several times. By that point in the time, the schema was already notified to update since it has it's own reference list to the field providers. at the end of the day this is easier and more efficient